scripts/classify-ci-failures.py

```
@@ -0,0 +1,752 @@
+
+def match_symptom(error_text: str) -> str | None:
+    for pattern, description in _FLAKY_SYMPTOM_PATTERNS:
+        if re.search(pattern, error_text, re.IGNORECASE):
+            return description
+    return None
+
+
+def _truncate_error(error: str, max_lines: int = 3, max_chars: int = 300) -> str:
+    if not error:
+        return ""
+    lines = [l for l in error.splitlines() if l.strip()][:max_lines]
```

Ruff (0.15.18)

[error] Ambiguous variable name: `l`

(E741)